### PR TITLE
{CosmosDB} az cosmosdb : Upgrade API version [2024-03-05] for Bug-Fix of InAccount Restore of Deleted Database/ Containers feature for SQL/Mongo/Table/Gremlin 

### PR DESCRIPTION
**Pull request by AgarwalVivek**
**Related command**

az cosmosdb sql container restore
az cosmosdb sql database restore
az cosmosdb mongodb collection restore
az cosmosdb mongodb database restore
az cosmosdb gremlin graph restore
az cosmosdb gremlin database restore
az cosmosdb table restore

**Description**
Fix #28365
Fixing a bug introduced with support for No TimeStamp Based Restore in release 2023-11-15: PR 28034

These CLI commands are for restoring deleted resources (Database / Collection) of cosmosdb account in the same live account.

**Testing Guide**

**History Notes**
[Cosmos DB] `az cosmosdb sql database/container restore`: Fix support for restore of deleted database resource in the same SQL account
[Cosmos DB] `az cosmosdb mongodb database/collection restore`: Fix support for restore of deleted database resource in the same MongoDB account
[Cosmos DB] `az cosmosdb gremlin database/graph restore`: Fix support for restore of deleted database resource in the same gremlin account
[Cosmos DB] `az cosmosdb table restore`: Fix support for restore of deleted table resource in the same account

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by evelyn-ys — CHANGES_REQUESTED** [review]
Other than three requirements.txt, you need to update the version in `setup.py`

**Comment by evelyn-ys** [comment]
Please also format PR title and description following https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-title

File: src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py [hunk 1]
```diff
@@ -3389,28 +3383,26 @@ def cli_cosmosdb_table_restore(cmd,
...
-                if resource.operation_type == "Delete" and latest_restore_time < event_timestamp and resource.owner_id == table_name:
-                    latest_restore_time = event_timestamp
+            latest_table_delete_time, latest_table_create_or_recreate_time, table_rid = process_restorable_databases(restorable_tables, table_name)  # pylint: disable=unused-variable
 
-            if (latest_restore_time == datetime.datetime.utcfromtimestamp(0)):
-                raise CLIError("Cannot find a deleted table with name {} in this account with name {} that is online in location {}".format(table_name, account_name, restorable_database_account.location))
+            # Table is alive if create or recreate timestamp is later than latest delete timestamp
+            table_alive = latest_table_create_or_recreate_time > latest_table_delete_time or latest_table_delete_time == datetime.datetime.max
+
+            if (table_alive):
```
Comment (amisi01) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:3398]: please correct the error message and if syntax similar to sql container or others.

Comment (AgarwalVivek) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py]: thanks for pointing out @amisi01. have updated code.

**Comment by evelyn-ys** [comment]
There seems no distribution for `azure-mgmt-cosmosdb==9.5.0`. See https://pypi.org/project/azure-mgmt-cosmosdb/#history

Can you pls make sure the new version has been released and the version is `9.5.0`?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**